crates/stackable-webhook/src/tls.rs

```diff
@@ -94,37 +97,43 @@ pub struct TlsServer {
 impl TlsServer {
     #[instrument(name = "create_tls_server", skip(router))]
     pub async fn new(socket_addr: SocketAddr, router: Router) -> Result<Self> {
-        let mut certificate_authority =
-            CertificateAuthority::new_rsa().context(CreateCertificateAuthoritySnafu)?;
-
-        let leaf_certificate = certificate_authority
-            .generate_rsa_leaf_certificate("Leaf", "webhook", Duration::from_secs(3600))
-            .context(GenerateLeafCertificateSnafu)?;
-
-        let certificate_der = leaf_certificate
-            .certificate_der()
-            .context(EncodeCertificateDerSnafu)?;
-
-        let private_key_der = leaf_certificate
-            .private_key_der()
-            .context(EncodePrivateKeyDerSnafu)?;
-
-        let tls_provider = default_provider();
-        let mut config = ServerConfig::builder_with_provider(tls_provider.into())
-            .with_protocol_versions(&[&TLS12, &TLS13])
-            .context(SetSafeTlsProtocolVersionsSnafu)?
-            .with_no_client_auth()
-            .with_single_cert(vec![certificate_der], private_key_der)
-            .context(InvalidTlsPrivateKeySnafu)?;
-
-        config.alpn_protocols = vec![b"h2".to_vec(), b"http/1.1".to_vec()];
-        let config = Arc::new(config);
-
-        Ok(Self {
-            socket_addr,
-            config,
-            router,
+        let task = tokio::task::spawn_blocking(move || {
```

Comment:
suggestion: Let's add a comment here, why we use the `spawn_blocking` method. And let's also add the link you provided in the PR description here.

Reply:
Yeah boi

Reply:
Done in f6902446c614ad83e9c3ceec78d36df66b09f279